spec/support/spec_test_helper.rb

```diff
@@ -6,6 +6,40 @@ def login_as(user)
   def login
     login_as(create(:user))
   end
+
+  RSpec::Matchers.define :match_hash do |expected|
+    match do |actual|
+      return false unless expected.is_a?(Hash) && actual.is_a?(Hash)
+      @actual = transform(actual)
+      @expected = transform(expected)
+      @actual.eql?(@expected)
+    end
+
+    def transform(parameter)
+      parameter.transform_keys!(&:to_s)
+      parameter.transform_values! { |v| transform_element(v) }
+      parameter.sort_by { |k, _v| k }.to_h
+    end
+
+    def transform_element(ele)
+      if ele.is_a?(ActiveRecord::Relation)
+        ele = ele.ordered if ele.respond_to?(:ordered)
+        ele.to_a
+      elsif ele.is_a?(Array)
+        ele = ele.sort unless ele[0].is_a?(Hash)
+        ele.map { |e| transform_element(e) }
+      elsif ele.is_a?(Hash)
+        transform(ele)
+      elsif ele.is_a?(ActiveRecord::Base)
+        ele
+      else
+        ele.to_s
```

Comment:
I guess we might need to edit this to support timestamps based on the current diff we're seeing. I'd prefer the format with the `T`, especially if that's the one we're currently returning, since it's RFC 3339 compliant, and I think datetime.to_s doesn't follow that by default - not sure.